### PR TITLE
CUMULUS-1816: URL Params reflect Datepicker on initial load

### PR DESCRIPTION
**Pull request by bt6453**
So Far:
* Fix string in Datepicker to use constant
* Update Datepicker to add timestamp to url params on initial page load

TODO:
* Is it ok that the timestamp gets changed to "Custom" when navigating to another page?
* Update Datepicker to only be 'Recent' timeframe on home page load

### CODE REVIEW

**Comment by bt6453** [comment]
Updated so that it will only automatically adjust to the "Recent" timeframe on home page load. Every other page will default to "All" if no url parameters are set.

**Comment by bt6453** [comment]
Ah ok. I misunderstood our earlier discussion then. I'll look into making it a one time thing on log in.

**Comment by flamingbear** [comment]
for the questions: 
"Is it ok that the timestamp gets changed to "Custom" when navigating to another page?" - it would be better if it didn't, but doesn't seem like it's worth too much effort if it's not easy to implement. 

"Update Datepicker to only be 'Recent' timeframe on home page load" - Yes, sorry I didnt even see this question. That's what I was talking about I think.

File: cypress/integration/collections_spec.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ describe('Dashboard Collections Page', () => {
 
     it('should display a link to view collections', () => {
       cy.contains('nav li a', 'Collections').as('collections');
-      cy.get('@collections').should('have.attr', 'href', '/collections');
+      cy.get('@collections').should('have.attr', 'href').and('match', /\/collections/);
```
Comment (flamingbear) [inline cypress/integration/collections_spec.js:41]: Is it possible to add `?startDateTime` to this match, just so we know to expect it?  NBD if not.

File: app/src/js/components/Datepicker/Datepicker.js [hunk 2]
```diff
@@ -63,6 +63,7 @@ class Datepicker extends React.PureComponent {
 
   componentDidMount () {
     updateDatepickerStateFromQueryParams(this.props);
+    this.homePageInitialDateTime();
```
Comment (flamingbear) [inline app/src/js/components/Datepicker/Datepicker.js:66]: I think this is probably going to get removed.

File: cypress/integration/main_page_spec.js [hunk 3]
```diff
@@ -281,5 +281,13 @@ describe('Dashboard Home Page', () => {
...
     });
+
+    it('adds the default datepicker options to the URL', () => {
+      const now = Date.UTC(2009, 0, 5, 13, 35, 3);
+      cy.clock(now);
+
+      cy.visit('/');
+      cy.url().should('include', 'startDateTime=20090104133500');
+    });
   });
```
Comment (flamingbear) [inline cypress/integration/main_page_spec.js:292]: We should also have the test where you visit('/'), "clear all" then click to granules (not "visit"), then click back to home and ensure the url is free of datetime QPs.

File: cypress/integration/collections_spec.js (hunk 1 above)
Comment (flamingbear) [reply cypress/integration/collections_spec.js:41]: Maybe only update one, because they're on all the pages.

File: test/reducers/datepicker.js [hunk 4]
```diff
@@ -16,16 +16,16 @@ test('verify initial state', (t) => {
   t.deepEqual(actual, inputstate);
 });
 
-test('reducer sets initial state to "Recent" on first initialization run.', (t) => {
-  const testStart = Date.now();
-  sinon.useFakeTimers(testStart);
+test('reducer sets initial state to "Custom" on first initialization run.', (t) => {
```
Comment (flamingbear) [inline test/reducers/datepicker.js:19]: I don't think this is true anymore.  It should just determine the label from the props I think.(?)

File: cypress/integration/providers_spec.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ describe('Dashboard Providers Page', () => {
 
     it('should display a link to view providers', () => {
```
Comment (flamingbear) [inline cypress/integration/providers_spec.js:31]: Can we add a test in one of the specs where we navigate directly to a "bookmarked url" and then verify that the Datepicker on that page reflects the state of the input URL?

File: app/src/js/components/Datepicker/Datepicker.js (hunk 2 above)
Comment (bt6453) [reply app/src/js/components/Datepicker/Datepicker.js:66]: I updated it a little bit, but should it be combined with `updateDatepickerStateFromQueryParams`? Although maybe it makes sense to keep their functions separate

File: cypress/integration/collections_spec.js (hunk 1 above)
Comment (bt6453) [reply cypress/integration/collections_spec.js]: With the new one of doing this, I was able to revert back most tests that had to be changed, but updated this for Collections spec

File: cypress/integration/main_page_spec.js (hunk 3 above)
Comment (bt6453) [reply cypress/integration/main_page_spec.js:327]: Added

File: cypress/integration/providers_spec.js (hunk 5 above)
Comment (bt6453) [reply cypress/integration/providers_spec.js]: Added

File: test/reducers/datepicker.js (hunk 4 above)
Comment (bt6453) [reply test/reducers/datepicker.js]: Yeah I think I worded this wrong for what I'm trying to test, which is just the initial state. Changed the title and cleaned it up.

File: app/src/js/components/Datepicker/Datepicker.js (hunk 2 above)
Comment (flamingbear) [reply app/src/js/components/Datepicker/Datepicker.js:66]: This is fine.

File: cypress/integration/main_page_spec.js (hunk 3 above)
Comment (flamingbear) [reply cypress/integration/main_page_spec.js:327]: sorry totally missed that in the last go around.

File: cypress/integration/main_page_spec.js [hunk 6]
```diff
@@ -281,5 +281,48 @@ describe('Dashboard Home Page', () => {
...
+      cy.get('[data-cy=datetime-clear]').click();
+      cy.contains('nav li a', 'Collections').as('collections');
+      cy.get('@collections').click();
+      cy.get('.logo').click();
+
+      cy.url().should('not.include', 'startDateTime=');
+    });
+
+    it('should update the Datepicker with the params in the URL', () => {
+      cy.visit('/?startDateTime=20081229133500&endDateTime=20090105133500');
```
Comment (flamingbear) [inline cypress/integration/main_page_spec.js:309]:
```suggestion
      cy.visit('/granules/?startDateTime=20081229133500&endDateTime=20090105133500');
```
This way we're testing an additional page beyond the homepage.  Take it or leave it, but I like it.

Comment (bt6453) [reply cypress/integration/main_page_spec.js:309]: Sure, that works.

**Pull request merged**